core/cmd/hoverfly/main.go

```diff
@@ -283,6 +282,8 @@ func main() {
 		requestCache = nil
 	}
 
+	cfg.DisableProxyAuthoriation = *disableProxyAuth
```

Comment:
The only options there are regarding authorization are `-auth` to turn on authentication and `-disable-standard-proxy-auth` to disable the standard Proxy-Authentication header. 

Trying to repurpose the `-auth` flag to be both a switch for authentication and which headers are used would be more confusing as it would mean that every time you want to turn on auth, you would have to specify which headers you want, as not providing a value would have the same meaning as not providing the flag.

```
hoverfly -auth -username benjih -password 123
```
The `-auth` flag would equal `""`

```
hoverfly
```
The `-auth` flag would also equal `""`

I think what you are proposing would be more confusing for standard use cases in which the user is operating in a secure network. Considering I think that this is a use case we only expect to occur for us when starting instances of Hoverfly in Hoverfly Cloud, I think what I have done is fine. 

The alternative would be forcing users to always start Hoverfly by specifying proxyAuth.
```
hoverfly -auth=proxyAuth -username benjih -password 123
```